### PR TITLE
[ONNX] Adds Support for Some Bitwise Ops in Onnx Exporter

### PR DESCRIPTION
**Pull request by zjgarvey**
Addresses #126194 

Adds support for 
- "aten::bitwise_right_shift"
- "aten::bitwise_left_shift"
- "aten::bitwise_and"

### CODE REVIEW

**Comment by zjgarvey** [comment]
To be completely honest, I'm not very familiar with the internals of torch.onnx.export. 

Are decomposition passes applied to a graph before converting torch fx nodes to onnx nodes? I also believe there is some automatic dtype promotion that occurs, but I got errors when leaving out the explicit type promotion in the conversion for "aten::bitwise_and". 

In any case, let me know if there is a better way of adding support for these ops and I can make changes.

**Comment by zjgarvey** [comment]
> I install the official torch==2.3.0 using pip , the number of lines in my code for symbolic_opset18.py is much less than the [one](https://github.com/pytorch/pytorch/blob/main/torch/onnx/symbolic_opset18.py) in current main branch,( only defining col2im) and then I added the changes from your two code files but encountered the following error. Do you know what I should do?

@JeremyGe07 With onnx opset version 18, onnx.ReduceMean switched from using axes as an attribute to using axes as an optional input tensor. See [onnx docs](https://onnx.ai/onnx/operators/onnx__ReduceMean.html#l-onnx-doc-reducemean). You could try exporting with an opset version < 18 (as long as you don't need the support from newer opset versions). 

Another thing that might be helpful if you want the most recent changes is to try using a nightly build of torch with 

```pip install torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html```

I'd recommend submitting an issue for the ReduceMean problem, since this PR is likely not the best place to discuss this.

**Comment by justinchuby** [comment]
If you intend to export fx graphs to ONNX. I recommend using the `torch.onnx.dynamo_export` api. Bitwise ops should already be supported

**Comment by justinchuby** [comment]
The pr itself looks good. Some tests may be preferable.

**Review by justinchuby — APPROVED** [review]

**Comment by zjgarvey** [comment]
> The pr itself looks good. Some tests may be preferable.

Good point. I'm happy to add some tests (particularly for `aten::bitwise_and`) if you would like.

File: torch/onnx/symbolic_opset18.py [hunk 1]
```diff
@@ -35,6 +35,30 @@
 _onnx_symbolic = functools.partial(registration.onnx_symbolic, opset=18)
 
 
+@_onnx_symbolic("aten::bitwise_and")
```
Comment (titaiwangms) [inline torch/onnx/symbolic_opset18.py:38]: Expose op to line 31

Comment (titaiwangms) [reply torch/onnx/symbolic_opset18.py:38]: Same for the other two ops

**Comment by zjgarvey** [comment]
I have a fix for most of the CI issues, but I'm not sure how to resolve the docs error:

```
undocumented objects found:
+ cat build/coverage/python.txt
Undocumented Python objects
===========================
torch.onnx.symbolic_opset18
---------------------------
Functions:
 * bitwise_and

+ echo 'Make sure you'\''ve updated relevant .rsts in docs/source!'
Make sure you've updated relevant .rsts in docs/source!
+ echo 'You can reproduce locally by running '\''cd docs && make coverage && cat build/coverage/python.txt'\'''
You can reproduce locally by running 'cd docs && make coverage && cat build/coverage/python.txt'
```

I'm able to reproduce the error, but I don't know what .rst to edit. Can someone give me advice on this?

**Comment by titaiwangms** [comment]
> I'm able to reproduce the error, but I don't know what .rst to edit. Can someone give me advice on this?

I thnk it should be this part that we have to add the new op that we use: https://github.com/pytorch/pytorch/blob/85fd76f76da7ebc95aed1e36b586649f091a1263/torch/onnx/symbolic_opset18.py#L7-L17

I reference on this: https://github.com/pytorch/pytorch/pull/118828/files#diff-4e77adb023a4e15f75009c08294251dfe9c3b7bb6947e7f7b4609c65edb53b20

**Comment by titaiwangms** [comment]
@pytorch merge

**Comment by titaiwangms** [comment]
@zjgarvey Thank you!

**Pull request closed without merging**